docs/source/includes/_structures.md

```diff
@@ -0,0 +1,340 @@
+</aside>
+
+
+## ChatMessage
+```json
+{
+  "message_id": 9150005004,
+  "sender_id": 2,
+  "channel_id": 5,
+  "timestamp": "2018-07-06T06:33:34+00:00",
+  "content": "i am a lazerface",
+  "is_action": 0,
+  "sender": {
+    "id": 2,
+    "username": "peppy",
+    "profile_colour": "#3366FF",
+    "avatar_url": "https://a.ppy.sh/2?1519081077.png",
+    "country_code": "AU",
+    "is_active": true,
+    "is_bot": false,
+    "is_online": true,
+    "is_supporter": true
+  }
+}
+```
+
+Represents an individual Message within a [ChatChannel](#chatchannel).
+
+Field      | Type                         | Description
+---------- | ---------------------------- | ------------------------------------------------------------
+message_id | number                       | unique identifier for message
+sender_id  | number                       | `user_id` of the sender
+channel_id | number                       | `channel_id` of where the message was sent
+timestamp  | string                       | when the message was sent, ISO-8601
+content    | string                       | message content
+is_action  | boolean                      | was this an action? i.e. `/me dances`
+sender     | [UserCompact](#usercompact)  | embeded UserCompact object to save additional api lookups
+
+
+## Notification
+```json
+{
+  "id": 1,
+  "name": "channel_message",
+  "created_at": "2019-04-24T07:12:43+00:00",
+  "object_type": "channel",
+  "object_id": 1,
+  "source_user_id": 1,
+  "is_read": true,
+  "details": {
+    "username": "someone",
+    ...
+  }
+}
+```
+
+Represents a notification object.
+
+Field            | Type    | Description
+---------------- | ------- | ------------------------------------------------------------------------
+id               | number  |
+name             | string  | Name of the event
+created_at       | string  | ISO 8601 date
+object_type      | string  |
+object_id        | number  |
+source_user_id   | number? |
+is_read          | boolean |
+details          | object  | `message_id` of last known message (only returned in presence responses)
+
+### Event Names
+
+Name                           | Description
+------------------------------ | -----------------------------------------------------
+beatmapset_discussion_lock     | Discussion on beatmap has been locked
+beatmapset_discussion_post_new | New discussion post on beatmap
+beatmapset_discussion_unlock   | Discussion on beatmap has been unlocked
+beatmapset_disqualify          | Beatmap was disqualified
+beatmapset_love                | Beatmap was promoted to loved
+beatmapset_nominate            | Beatmap was nominated
+beatmapset_qualify             | Beatmap has gained enough nominations and entered the ranking queue
+beatmapset_reset_nominations   | Nomination of beatmap was reset
+channel_message                | Someone sent chat message
+forum_topic_reply              | Someone replied on forum topic
+
+### `beatmapset_discussion_lock`
```

Comment:
No examples for these notification objects?

Also, might be worth grouping these by `object_type` or something to make it easier to follow... right now it's a wall of... tables?

Reply:
To be done later™.